src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -69,7 +69,7 @@ public class MultiSyncModeSelector : ISyncModeSelector, IDisposable
 
         private readonly long _pivotNumber;
         private bool FastSyncEnabled => _syncConfig.FastSync;
-        private bool SnapSyncEnabled => _syncConfig.SnapSync && !_isSnapSyncDisabledAfterAnyStateSync;
+        private bool SnapSyncEnabled => _syncConfig.SnapServe || (_syncConfig.SnapSync && !_isSnapSyncDisabledAfterAnyStateSync);
```

Comment:
Must snapsync be enabled in order to serve it?

Reply:
technically no, but I did this thinking of SnapSync=true as a flag that enables the snap protocol in general, then SnapServe=true enables serving.

Reply:
i think since `MultiSyncModeSelector` is just for syncing, we should not introduce anything related to serving here. And anyways this `SnapSyncEnabled` flag does not play any role in enabling/disabling snap protocol. I should restore it.